src/lib/validate-input.js

```diff
@@ -611,6 +611,44 @@ export function validateGatherParams(input, indices, {axis = 0} = {}) {
   }
 }
 
+export function validateScatterElementsParams(input, indices, updates, {axis = 0} = {}) {
+  const inputRank = input.rank;
+  const indicesRank = indices.rank;
+  const updatesRank = updates.rank;
+  if (inputRank < 1) {
+    throw new Error(`The input should be at least a 1-D tensor.`);
+  }
+  if (indicesRank < 1) {
+    throw new Error(`The indices should be at least a 1-D tensor.`);
+  }
+  if (updatesRank < 1) {
+    throw new Error(`The updates should be at least a 1-D tensor.`);
+  }
+  if (indicesRank !== inputRank) {
+    throw new Error(`Invalid indices value - indices rank should be equal to input rank.`);
+  }
+  if (updatesRank !== indicesRank) {
+    throw new Error(`Invalid updates value - updates rank should be equal to indices rank.`);
+  }
+  if (!updates.shape.every((value, index) => value === indices.shape[index])) {
+    throw new Error(`Invalid updates value - updates shape should be same as indices shape.`);
+  }
+  if (axis !== undefined) {
+    if (!Number.isInteger(axis) || axis < 0 || axis >= inputRank) {
+      throw new Error(
+          `The axis ${axis} should be an unsigned integer in the interval [0, ${inputRank}).`);
+    }
+  }
```

Comment:
Should we set default axis value if axis === undefined?

Reply:
Thanks Ningxin!
I've already set default `axis` as `0` when defining `scatterElements(input, indices, updates, {axis = 0} = {})` function. So  above `if` condition - `if (axis !== undefined)` is always `true`, no need to check that. I removed it in new commit.